node/src/sub_lib/neighborhood.rs

```diff
@@ -478,157 +536,248 @@ mod tests {
     }
 
     #[test]
-    fn try_dismantle_works_for_ethereum_mainnet() {
-        let descriptor = "masq://eth.as45cs5c5:1.2.3.4:4444";
+    fn parse_works_for_ethereum_mainnet() {
+        let descriptor = "masq://eth-mainnet:as45cs5c5@1.2.3.4:4444";
 
-        let result = NodeDescriptor::try_dismantle_str(descriptor).unwrap();
+        let result = NodeDescriptor::parse_url(descriptor).unwrap();
 
-        assert_eq!(
-            result,
-            (Blockchain::EthMainnet, "as45cs5c5", "1.2.3.4:4444")
-        )
+        assert_eq!(result, (Chain::EthMainnet, "as45cs5c5", "1.2.3.4:4444"))
     }
 
     #[test]
-    fn try_dismantle_str_works_for_ethereum_testnet_ropsten() {
-        let descriptor = "masq://eth_t1.as45cs5c5:1.2.3.4:4444";
+    fn parse_works_for_ropsten() {
+        let descriptor = "masq://eth-ropsten:as45cs5c5@1.2.3.4:4444";
```

Comment:
Is this 'eth-ropsten' a segment that will be passed to `Chain::from`? It uses just 'ropsten' to match, correct?

Reply:
Not exactly this is attached to a function which expect only the long (full length) identifiers because they are what we insist on as the part of the descriptor. 

It's called pub `fn chain_from_chain_identifier_opt(identifier: &str) -> Option<Chain>`
so it's different from Chain::from(). I think I don't want to change this. It allows me to be more strict about the syntax, despite the loosen boundaries fro in-command names.